src/components/utilities/Modal.tsx

```diff
@@ -0,0 +1,115 @@
+import React from 'react';
+import reactDom from 'react-dom';
+import styled from 'styled-components';
+import closeBtnImg from '../../assets/Popup-close-button.svg';
+
+interface Props {
+  title?: string;
+  onClose: () => void;
+  children: React.ReactNode;
+}
+
+// Styled component for backdrop
+const Backdrop = styled.div`
+  background: rgba(0, 0, 0, 0.25);
+  position: absolute;
+  top: 0;
+  left: 0;
+  width: 100vw;
+  height: 100vh;
+  z-index: 999 !important;
```

Comment:
Is it good practice to do `999 !important` here? I understand you're just making sure the backdrop is on top but I'm not sure if this is the generally accepted way to do it

Reply:
Thanks for pointing this out and I also think this isn't a good way. I found a fix to avoid z-index by using browser's default stacking order. I have made a new commit now..

Reply:
I initially used z-index because i placed the modal component directly inside Programs component, but this caused nesting and i think its not good for accessibility too.. so i used React Portal because of this. However still had to use z-index because the main navbar had position=fixed and it was stacking on top of the backdrop...

Reply:
But turns out i just had to reshuffle the root elements in the index.html so that it stacks up the way we want

Reply:
Thanks for fixing this, cool that you figured it out!